user-guide/Cloud_Platform/Catalog/Tutorials/Tutorial_Register_Catalog_Version_GitHub_Actions.md

```diff
@@ -0,0 +1,133 @@
+
+## Step 1: Create the GitHub Actions Workflow file
+
+- Go to your GitHub Repository.  
+    Open the repository where you want to add this GitHub Actions workflow.
+
+- Create a .github/workflows Directory.  
+    In the root of your repository, create a directory named `.github` (if it doesn't exist).
+    Inside the `.github` directory, create another directory called `workflows`.
+
+- Add the Workflow File.  
+Inside the `workflows` directory, create a new file named `dataminer-catalog-pipeline.yml`.
+Paste below pipeline template into this file.
+
+```yaml
+name: Build and Register a Catalog version
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build_and_upload:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the repository
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Install the Skyline DataMiner CICD Packager Tool
+      - name: Install Skyline DataMiner CICD Packager Tool
+        run: dotnet tool install -g Skyline.DataMiner.CICD.Tools.Packager
+
+      # Create the DataMiner Protocol package
+      - name: Create DataMiner Protocol Package
+        run: dataminer-package-create dmprotocol "${{ github.workspace }}" --name catalog_registration_tutorial --output "${{ github.workspace }}/Packages"
+
+      # Get the catalog ID (must be set in the repository secrets)
+      - name: Get Catalog ID
+        run: echo "Catalog ID: ${{ secrets.CATALOG_ID }}"
+      
+      # Prepare the package file and version details
+      - name: Register on Catalog
+        run: |
+          $file = Get-Item "${{ github.workspace }}/Packages/catalog_registration_tutorial.dmprotocol"
+          $uri = "https://api.dataminer.services/api/key-catalog/v1-0/catalog/${{ secrets.CATALOG_ID }}/register/version"
+          
+          # Define the form data (package file, version, and description)
+          $formData = @{
+            file              = Get-Item $file.FullName
+            versionNumber     = "${{ secrets.VERSION_NUMBER }}"
+            versionDescription = "New version registered by GitHub Actions pipeline"
+          }
+
+          # Set up the API request to register the new version in the catalog
+          Invoke-RestMethod -Uri $uri -Method Post -Headers @{ 'Ocp-Apim-Subscription-Key' = "${{ secrets.API_TOKEN }}" } -Form $formData
+```
+
+## Step 2: Add GitHub secrets
+
+To securely store sensitive information like the catalog ID, API token, and version number, follow these steps:
+
+- Navigate to GitHub Secrets.  
+    In your GitHub repository, go to the repository's main page.  
+    Click on the `Settings` tab.  
+    On the left sidebar, click on `Secrets and variables` > `Actions`.  
+    Click the `New repository secret` button.
+
+
+- Add the Required Secrets.  
+`CATALOG_ID`: This is the catalog ID. Make sure this is the correct id of the [Catalog item](xref:Tutorial_Register_Catalog_Item) you registered earlier.   
```

Comment:
this should probably not be a secret?

Reply:
changed it to using the github run number in version number

Reply:
But this does not change the fact that the CatalogID still remains a secret right? Can we not make this a variable or just add it in this file plain text?

Reply:
i read your comment too fast :) fixed it for version number only instead of for catalog Id. will adapt

Reply:
updated to use an env var

Reply:
proabably still need to remove it now from being added to the secrets in the guide? line 102

Reply:
indeed, removed now